osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs

```diff
@@ -52,11 +56,12 @@ public partial class BeatmapPicker : Container
             }
         }
 
-        public BeatmapPicker()
+        public BeatmapPicker(UserSquareList userSquareList)
```

Comment:
I don't think you need to pass this around everywhere. With some position adjustments, something like this should work:

```diff
diff --git a/osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs b/osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs
index 02cbf8c2fb..54dcd5b289 100644
--- a/osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs
+++ b/osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs
@@ -56,12 +56,17 @@ public partial class BeatmapPicker : Container
             }
         }
 
-        public BeatmapPicker(UserSquareList userSquareList)
+        public BeatmapPicker()
         {
             RelativeSizeAxes = Axes.X;
             AutoSizeAxes = Axes.Y;
+
             favourites = new Statistic(FontAwesome.Solid.Heart);
-            usersList = userSquareList;
+            usersList = new UserSquareList
+            {
+                BypassAutoSizeAxes = Axes.Both,
+            };
+
             Children = new Drawable[]
             {
                 new FillFlowContainer
@@ -144,6 +149,7 @@ public BeatmapPicker(UserSquareList userSquareList)
                         },
                     },
                 },
+                usersList
             };
 
             Beatmap.ValueChanged += b =>
diff --git a/osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs b/osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs
index 2e7fc7016e..11c0434962 100644
--- a/osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs
+++ b/osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs
@@ -24,7 +24,6 @@
 using osu.Game.Online.API.Requests.Responses;
 using osu.Game.Online.Chat;
 using osu.Game.Overlays.BeatmapSet.Buttons;
-using osu.Game.Users;
 using osuTK;
 using osuTK.Graphics;
 
@@ -69,7 +68,6 @@ public BeatmapSetHeaderContent()
         {
             RelativeSizeAxes = Axes.X;
             AutoSizeAxes = Axes.Y;
-            var recentFavouritedUsersList = new UserSquareList();
             InternalChild = new Container
             {
                 RelativeSizeAxes = Axes.X,
@@ -115,7 +113,7 @@ public BeatmapSetHeaderContent()
                                     {
                                         RelativeSizeAxes = Axes.X,
                                         AutoSizeAxes = Axes.Y,
-                                        Child = Picker = new BeatmapPicker(recentFavouritedUsersList),
+                                        Child = Picker = new BeatmapPicker(),
                                     },
                                     title = new MetadataFlowContainer(s =>
                                     {
@@ -188,7 +186,6 @@ public BeatmapSetHeaderContent()
                             Details = new Details(),
                         },
                     },
-                    recentFavouritedUsersList
                 },
             };
 

```